### PR TITLE
Jh/enhanced mdx

### PR DESCRIPTION
**Pull request by joelhooks**

### CODE REVIEW

File: src/layouts/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+export default function Layout(frontMatter) {
```
Comment (joelhooks) [inline src/layouts/index.tsx:3]: this was throwing me off. templates are functions that return a component function and deliver frontmatter

File: src/pages/index.mdx [hunk 2]
```diff
@@ -1,8 +1,12 @@
-<div className="w-64"><img src="/images/eggo.svg" /></div>
+<div className="w-64">
```
Comment (joelhooks) [inline src/pages/index.mdx:1]: with templates we can just use mdx instead of a wrapper

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import {useURL} from '@hooks/useUrl'
+import '../styles/index.css'
+
+export default class App extends NextApp {
+  render() {
+    const {Component, pageProps} = this.props
+    return (
+      <>
+        <DefaultSeo {...defaultSeoConfig} />
+        <SocialProfileJsonLd
```
Comment (johnlindquist) [inline src/pages/_app.tsx:19]: NICE FIND

File: src/components/mdx/index.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Course from './course'
+import Link from 'next/link'
+
+const mdxComponents = {
+  Course,
+  Link,
```
Comment (johnlindquist) [inline src/components/mdx/index.ts:6]: Oh yah, good idea w/ Link

File: src/layouts/ultimate-guide.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    // React hooks, for example `useState` or `useEffect`, go here.
+
+    console.log(frontMatter)
+    return (
+      <>
+        <NextSeo
+          title={title}
+          description={description}
+          titleTemplate={titleAppendSiteName ? undefined : '%s'}
+          openGraph={{
```
Comment (johnlindquist) [inline src/layouts/ultimate-guide.tsx:21]: Very cool layout usage

**Pull request merged**